include/mbedtls/gcm.h

```diff
@@ -69,6 +69,7 @@ typedef struct mbedtls_gcm_context
     int mode;                             /*!< The operation to perform:
                                                #MBEDTLS_GCM_ENCRYPT or
                                                #MBEDTLS_GCM_DECRYPT. */
+    uint8_t ad_data_finished;
```

Comment:
a flag ment to be checked during first call to `mbedtls_gcm_update` or mbedtls_gcm_finish`.

Reply:
How about using `ctx->len==0`? This tells you whether `update` has been called with a nonzero length, which is slightly different (in case `update` is first called with length=0) but gives the same result in the end. By leveraging this existing information, you don't need the code to update the flag, just the code to finish authenticating the AD when `ctx->len == 0 && ctx->add_len % 16 1= 0` at the beginning of `update` or `finish`.

Reply:
ah! great! exactly what I needed :)